test/test_autograd.py

```diff
@@ -855,6 +855,20 @@ def hook(grad):
             self.assertEqual(x.grad, expected_grad)
             self.assertIsNone(x_list[i].grad)
 
+    def test_grad_zero_grad_unused(self):
+        x = torch.tensor(0.5, dtype=torch.float32, requires_grad=True)
+        a = torch.tensor(1, dtype=torch.float32, requires_grad=True)
+        y = x * a
+        dydx = torch.autograd.grad(y, x, create_graph=True, allow_unused=True, zero_grad_unused=True)
+        d2ydx2_None = torch.autograd.grad(dydx, x, create_graph=True, allow_unused=True)
+        d2ydx2 = torch.autograd.grad(dydx, x, create_graph=True, allow_unused=True, zero_grad_unused=True)
+        self.assertIsNone(d2ydx2_None[0])
+        self.assertEqual(d2ydx2[0], 0)
+        try:
+            d3ydx3 = torch.autograd.grad(d2ydx2, x, allow_unused=True, zero_grad_unused=True)
```

Comment:
You don't need a try/except for this if you don't want it to raise an error. The test should already fail if this line raises an error.

Reply:
This try-catch block was removed, with a new assert testing the value of `d3ydx3`.